src/nvim/event/process.c

```diff
@@ -423,6 +423,7 @@ static void exit_event(void **argv)
 
   if (!exiting) {
     if (ui_client_channel_id) {
+      ui_client_exit_status = status;
       os_exit(status);
```

Comment:
Off-topic: `os_exit` doesn't [set v:exiting](https://github.com/neovim/neovim/blob/a293e2794dcf65a35eda6eeea6ea480bfef9dc73/src/nvim/main.c#L697) , wonder if we can all `getout` here instead.

Reply:
This code runs in the TUI process, `v:exiting` and the stuff which happens in `getout()` is not relevant for it (it might work by accident, but I think using `os_exit()` directly expresses the intent).